logstash-core/lib/logstash/runner.rb

```diff
@@ -184,6 +184,11 @@ def run(args)
   end
 
   def execute
+    # Only when execute is have the CLI options been added to the @settings
+    # We invoke post_process to apply extra logic to them.
+    # The post_process callbacks have been added in environment.rb
+    @settings.post_process
```

Comment:
I wonder, instead of manually calling this in the `execute` if we could make the `#validate_all` call this before actually validating the value? I am thinking in a context of **universal plugin**, they should also be able to add post process callbacks.